universal_transfer_operator/transfers/datasets/base.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from urllib.parse import urlparse
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+from attr import define, field
+
+try:
+    # Airflow >= 2.4
+    from airflow.datasets import Dataset
+
+    DATASET_SUPPORT = True
+except ImportError:
+    # Airflow < 2.4
+    Dataset = object
+    DATASET_SUPPORT = False
+
+
+@define
+class UniversalDataset(LoggingMixin, Dataset):
```

Comment:
Where and why do we need this?

Reply:
File, Table, Dataframe and API classes are common methods that would be part of UniversalDataset. UniversalTransferOperator uses this UniversalDataset only like https://github.com/astronomer/astro-sdk/blob/bc459fa8756fb2612b6a064520a31995dac49c1d/universal_transfer_operator/transfers/universal_transfer_operator.py#L8

Reply:
Removed this now as uri is not being passed by user: https://github.com/astronomer/astro-sdk/pull/1492/commits/0f409c31eed44985e27326c89164b5bdbf7f0284